### PR TITLE
ur: remove anyhow as a dependency.

### PR DESCRIPTION
**Pull request by jeandudey**
I'm a bit confused about the casing, do you prefer starting with a capital letter?

### CODE REVIEW

File: src/fountain.rs [hunk 1]
```diff
@@ -82,6 +82,53 @@
 //! );
 //! ```
 
+use std::convert::Infallible;
+
+/// Errors that can happen during fountain encoding and decoding.
+#[derive(Debug)]
+pub enum Error {
+    /// CBOR decodee  error.
```
Comment (dspicher) [inline src/fountain.rs:90]:
```suggestion
    /// CBOR deconding error.
```

File: src/ur.rs [hunk 2]
```diff
@@ -26,7 +26,48 @@
 //! assert_eq!(decoder.message().unwrap().as_deref(), Some(data.as_bytes()));
 //! ```
 
-use anyhow::Context;
+/// Errors that can happen during encoding and decoding of UR.
```
Comment (dspicher) [inline src/ur.rs:29]:
```suggestion
/// Errors that can happen during encoding and decoding of URs.
```

**Review by dspicher — CHANGES_REQUESTED** [review]
That's a very nice improvement, thanks for the PR!

It's fine to leave the error message casing inconsistent for now, we can clean it up later.

**Comment by jeandudey** [comment]
I've removed the `to_string` occurrences. Also took it one step further and made `hex` a dev-dependency to prepare to add no_std support.

**Comment by dspicher** [comment]
> I've removed the `to_string` occurrences. Also took it one step further and made `hex` a dev-dependency to prepare to add no_std support.

Nice idea! I found a bit a nicer way to do it, which doesn't involve a new struct: https://github.com/dspicher/ur-rs/pull/160

Feel free to give it a review, or include the diff here if you would like.

**Comment by jeandudey** [comment]
@dspicher I prefer your way. It's cleaner. Feel free to merge yours first and then I'll rebase

**Comment by dspicher** [comment]
> @dspicher I prefer your way. It's cleaner. Feel free to merge yours first and then I'll rebase

Done.

**Comment by jeandudey** [comment]
@dspicher rebased the PR

**Review by dspicher — APPROVED** [review]

**Pull request merged**